airflow/migrations/versions/0050_3_0_0_remove_pickled_data_from_dagrun_table.py

```diff
@@ -0,0 +1,221 @@
+        sa.Column("queued_at", sa.TIMESTAMP(timezone=True), nullable=True),
+        sa.Column("logical_date", sa.TIMESTAMP(timezone=True), nullable=False),
+        sa.Column("start_date", sa.TIMESTAMP(timezone=True), nullable=True),
+        sa.Column("end_date", sa.TIMESTAMP(timezone=True), nullable=True),
+        sa.Column("state", sa.String(length=50), nullable=True),
+        sa.Column("run_id", sa.String(length=250), nullable=False),
+        sa.Column("creating_job_id", sa.Integer(), nullable=True),
+        sa.Column("external_trigger", sa.Boolean(), nullable=True),
+        sa.Column("run_type", sa.String(length=50), nullable=False),
+        sa.Column("triggered_by", sa.String(length=50), nullable=True),
+        sa.Column("conf", sa.LargeBinary().with_variant(LONGBLOB, "mysql"), nullable=True),
+        sa.Column("data_interval_start", sa.TIMESTAMP(timezone=True), nullable=True),
+        sa.Column("data_interval_end", sa.TIMESTAMP(timezone=True), nullable=True),
+        sa.Column("last_scheduling_decision", sa.TIMESTAMP(timezone=True), nullable=True),
+        sa.Column("log_template_id", sa.Integer(), nullable=True),
+        sa.Column("updated_at", sa.TIMESTAMP(timezone=True), nullable=True),
+        sa.Column("clear_number", sa.Integer(), nullable=False, server_default=sa.text("0")),
+        sa.Column("backfill_id", sa.Integer(), nullable=True),
+        sa.Column("dag_version_id", UUIDType(binary=False), nullable=True),
+        sa.PrimaryKeyConstraint("id"),
+        sa.UniqueConstraint("dag_id", "run_id"),
+        if_not_exists=True,
+    )
+
+    # Condition to detect pickled data for different databases
+    condition_templates = {
+        "postgresql": "get_byte(conf, 0) = 128",
+        "mysql": "HEX(SUBSTRING(conf, 1, 1)) = '80'",
+        "sqlite": "hex(substr(conf, 1, 1)) = '80'",
+    }
+
+    condition = condition_templates.get(dialect)
+    if not condition:
+        raise RuntimeError(f"Unsupported dialect: {dialect}")
+
+    # Archive pickled data using the condition
+    conn.execute(
+        text(
+            f"""
+             INSERT INTO _dag_run_archive (dag_id, run_id, queued_at, logical_date, start_date, end_date, state, creating_job_id, external_trigger, run_type, triggered_by, conf, data_interval_start, data_interval_end, last_scheduling_decision, log_template_id, updated_at, clear_number, backfill_id, dag_version_id)
+    SELECT dag_id, run_id, queued_at, logical_date, start_date, end_date, state, creating_job_id, external_trigger, run_type, triggered_by, conf, data_interval_start, data_interval_end, last_scheduling_decision, log_template_id, updated_at, clear_number, backfill_id, dag_version_id
+    FROM  dag_run
+    WHERE conf IS NOT NULL AND {condition};
+
+                """
+        )
+    )
+
+    # Delete the pickled data from the dag_run table so that we can update the column type
+    conn.execute(text(f"DELETE FROM dag_run WHERE conf IS NOT NULL AND {condition}"))
```

Comment:
We don't want to delete the rows right? That'll cascade down and delete other stuff too, like TIs.

Should we just set `conf` to null instead?

Reply:
One difference with this situation vs with xcom is the xcom value is the "valuable" part of the row - this conf isn't though.

Reply:
Yes, this is better.
> Should we just set conf to null instead?

Reply:
> Yes, this is better.
> 
> > Should we just set conf to null instead?

This is done